test/core/call_indirect.wast

```diff
@@ -80,6 +80,38 @@
     ))
   )
 
+  (func
+    (block i32.const 0 call_indirect)
+    (loop i32.const 0 call_indirect)
+    (if (i32.const 0) (then i32.const 0 call_indirect))
+    (if (i32.const 0)
+      (then i32.const 0 call_indirect)
+      (else i32.const 0 call_indirect)
+    )
+    (block i32.const 0 i32.const 0 call_indirect (param i32))
```

Comment:
Might be good to test some with results too.

Reply:
Added more tests, including (type $t) and multiple fields.